src/lib/optimizer/strategy/projection_combination_rule.cpp

```diff
@@ -0,0 +1,106 @@
+#include "projection_combination_rule.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+
+namespace opossum {
+
+std::string ProjectionCombinationRule::name() const { return "Projection Combination Rule"; }
+
+bool ProjectionCombinationRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) {
+  if (node->type() != LQPNodeType::Projection) {
+    return _apply_to_children(node);
+  }
+
+  std::vector<std::shared_ptr<ProjectionNode>> projection_nodes;
+
+  // Gather subsequent ProjectionNodes
+  auto current_node = node;
+  while (current_node->type() == LQPNodeType::Projection) {
+    projection_nodes.emplace_back(std::static_pointer_cast<ProjectionNode>(current_node));
+
+    // If current node has multiple or no children, end the combination here
```

Comment:
ProjectionNodes will never have a right child and always have a left child, otherwise the LQP is invalid, so these check will never fire (At max, Assert for this.)

Reply:
I have replaced the checks with DebugAsserts now, since, based on that this case should not happen anyway, we are good with DebugAsserts over release Asserts.